src/plugins/data/public/query/query_string/language_service/lib/language_reference.tsx

```diff
@@ -7,11 +7,27 @@ import { i18n } from '@osd/i18n';
 
 import { EuiButtonIcon, EuiPopover, EuiPopoverTitle } from '@elastic/eui';
 
-import React, { ReactFragment } from 'react';
+import React, { ReactFragment, useEffect, useState } from 'react';
 import { FormattedMessage } from 'react-intl';
 
-export const LanguageReference = (props: { body: ReactFragment }) => {
-  const [isLanguageReferenceOpen, setIsLanguageReferenceOpen] = React.useState(false);
+export const LanguageReference = (props: {
+  body: ReactFragment;
+  autoShow?: boolean;
+  selectedLanguage?: string;
+}) => {
+  const [isLanguageReferenceOpen, setIsLanguageReferenceOpen] = useState(props.autoShow || false);
+
+  useEffect(() => {
+    if (props.autoShow) {
+      const storageKey = `hasSeenInfoBox_${props.selectedLanguage}`;
+      const hasSeenInfoBox = window.localStorage.getItem(storageKey);
+
+      if (hasSeenInfoBox === null && props.autoShow) {
```

Comment:
It doesn't seem to need check `props.autoShow` here again because it's checked on line 21
```suggestion
      if (hasSeenInfoBox === null) {
```